src/test/java/org/mskcc/cbio/oncokb/web/rest/SlackControllerIT.java

```diff
@@ -0,0 +1,647 @@
+        actionJSON.setResponseUrl(USER_REGISTRATION_WEBHOOK);
+
+        // Set payload token
+        actionJSON.setToken(DEFAULT_PAYLOAD_TOKEN);
+
+        return actionJSON;
+    }
+
+    private ViewSubmissionPayload getViewSubmissionPayload(DropdownEmailOption option) throws NullPointerException{
+        ViewSubmissionPayload viewJSON = new ViewSubmissionPayload();
+
+        viewJSON.setToken(DEFAULT_PAYLOAD_TOKEN);
+
+        List<ViewSubmissionPayload.ResponseUrl> responseUrls = new ArrayList<>();
+        ViewSubmissionPayload.ResponseUrl responseUrl = new ViewSubmissionPayload.ResponseUrl();
+        responseUrl.setResponseUrl(USER_REGISTRATION_WEBHOOK);
+        viewJSON.setResponseUrls(responseUrls);
+
+        ViewSubmissionPayload.User user = new ViewSubmissionPayload.User();
+        user.setName(DEFAULT_ADDR);
+        viewJSON.setUser(user);
+
+        View view = new View();
+        view.setPrivateMetadata(slackService.getOptionValue(USER_REGISTRATION_WEBHOOK, DEFAULT_USER_EMAIL));
+        view.setCallbackId(option.getConfirmActionId().orElseThrow(NullPointerException::new).getId());
+        ViewState viewState = new ViewState();
+        Map<String, Map<String, ViewState.Value>> values = new HashMap<>();
+        Map<String, ViewState.Value> subjectMap = new HashMap<>();
+        ViewState.Value subjectValue = new ViewState.Value();
+        subjectValue.setValue(messageSource.getMessage(option.getMailType().getTitleKey(), new Object[]{}, Locale.forLanguageTag(DEFAULT_LANG_KEY)));
+        subjectMap.put(ActionId.INPUT_SUBJECT.getId(), subjectValue);
+        Map<String, ViewState.Value> bodyMap = new HashMap<>();
+        ViewState.Value bodyValue = new ViewState.Value();
+        bodyValue.setValue(slackService.getStringFromResourceTemplateMailTextFile(option.getMailType().getStringTemplateName().orElseThrow(NullPointerException::new)));
+        bodyMap.put(ActionId.INPUT_BODY.getId(), bodyValue);
+        values.put(BlockId.SUBJECT_INPUT.getId(), subjectMap);
+        values.put(BlockId.BODY_INPUT.getId(), bodyMap);
+        viewState.setValues(values);
+        view.setState(viewState);
+        viewJSON.setView(view);
+
+        return viewJSON;
+    }
+
+    private void checkSlackBlock(String url, Payload payload, ActionId actionId, boolean collapsed) {
+        assertThat(url).isEqualTo(USER_REGISTRATION_WEBHOOK);
+
+        // Add required values
+        Set<String> expectedValues = new HashSet<>();
+        expectedValues.add(DEFAULT_USER_EMAIL);
+        expectedValues.add(DEFAULT_COMPANY_NAME);
+        if (collapsed) {
+            expectedValues.add(BlockId.COLLAPSED.getId());
+            expectedValues.add(actionId == ActionId.CHANGE_LICENSE_TYPE ? OTHER_LICENSE_TYPE.getShortName() : DEFAULT_LICENSE_TYPE.getShortName());
+            if (actionId == ActionId.GIVE_TRIAL_ACCESS) {
+                expectedValues.add("TRIAL");
+            } else if (ActionId.isConfirmEmailAction(actionId)) {
```

Comment:
Curious why we don't add the collapsed noted as long as the DropdownEmailOption has.

Reply:
Is it to avoid unnecessary for loop? The same goes to line 558-563

Reply:
Yes, it's mainly just so the cases that aren't dropdown emails don't have to go through the for loop. Definitely can omit and just add based on match though, is that better?